pkg/reconciler/openshift/tektonaddon/resolver_task.go

```diff
@@ -26,18 +26,44 @@ import (
 )
 
 func (r *Reconciler) EnsureResolverTask(ctx context.Context, enable string, ta *v1alpha1.TektonAddon) error {
-	manifest := *r.resolverTaskManifest
+	manifest := r.resolverTaskManifest
+	return r.ensureCustomSet(ctx, enable, ResolverTaskInstallerSet, ta, manifest, r.getTransformer(ctx, KindTask, false))
+}
+
+func (r *Reconciler) EnsureResolverStepAction(ctx context.Context, enable string, ta *v1alpha1.TektonAddon) error {
+	manifest := r.resolverStepActionManifest
+	return r.ensureCustomSet(ctx, enable, ResolverStepActionInstallerSet, ta, manifest, r.getTransformer(ctx, KindStepAction, false))
+}
+
+func (r *Reconciler) getTransformer(ctx context.Context, kind string, isVersioned bool) []mf.Transformer {
```

Comment:
instead of this function, we can just pass the tranformers as argument directly, i dont see much benefit of this func with switch case

Reply:
If you think, this is valid, handle in a follow-up PR.